### PR TITLE
feat(additional-properties): handle typed additional properties in model

### LINKED ISSUE #68: Add support for typed additional properties
Currently, there are no utility methods for serialization and deserialization of typed additional properties. This ticket comprises of the following items
 - Add/update utilities to handle json serialization, form serialization and json deserialization of typed additional properties
 - Add unit tests for the typed additional properties

### PR DESCRIPTION
**Pull request by sufyankhanrao**
## What
This PR adds the handling for the typed additional properties in serialization (form and json) and de-serialization. Also, contains the tests for the newly added changes.

## Why
 - To add support for the typed additional properties in the SDK

Closes #68 

## Type of change
Select multiple if applicable.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause a breaking change)
- [ ] Tests (adds or updates tests)
- [ ] Documentation (adds or updates documentation)
- [ ] Refactor (style improvements, performance improvements, code refactoring)
- [ ] Revert (reverts a commit)
- [ ] CI/Build (adds or updates a script, change in external dependencies)

## Dependency Change
If a new dependency is being added, please ensure that it adheres to the following guideline https://github.com/apimatic/apimatic-codegen/wiki/Policy-of-adding-new-dependencies-in-the-core-libraries

## Breaking change
If the PR is introducing a breaking change, please ensure that it adheres to the following guideline https://github.com/apimatic/apimatic-codegen/wiki/Guidelines-for-maintaining-core-libraries

## Testing
List the steps that were taken to test the changes

## Checklist
- [x] My code follows the coding conventions
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added new unit tests

### CODE REVIEW

File: apimatic_core/utilities/api_helper.py [hunk 1]
```diff
@@ -668,6 +676,27 @@ def to_lower_case(list_of_string):
...
+
+        Args:
+            dictionary (dict): The dictionary to extract additional properties from.
+            unboxing_function (callable): The deserializer to apply to each item in the dictionary.
+
+        Returns:
+            dict: A dictionary containing the additional properties and their values.
+        """
+        additional_properties = {}
+        for key, value in dictionary.items():
```
Comment (MaryamAdnan3) [inline apimatic_core/utilities/api_helper.py:691]: How will the unboxing function will behave incase of value as array or dictionary given we are passing the function **lambda x: str(x))**

File: apimatic_core/utilities/api_helper.py [hunk 2]
```diff
@@ -668,6 +676,27 @@ def to_lower_case(list_of_string):
...
+            try:
+                additional_properties[key] = unboxing_function(value)
+            except Exception:
```
Comment (MaryamAdnan3) [inline apimatic_core/utilities/api_helper.py:694]: we will ignore the values that will produce any exception, is this behaviour consistent accross all languages

Comment (sufyankhanrao) [reply apimatic_core/utilities/api_helper.py:694]: yes

File: apimatic_core/utilities/api_helper.py (hunk 1 above)
Comment (sufyankhanrao) [reply apimatic_core/utilities/api_helper.py:691]: it is already known like what will be the type of the additional property values, therefore I handled it accordingly.

**Review by MaryamAdnan3 — DISMISSED** [review]
Please merge after the build passes, The implementation looks good to me.

**Review by MaryamAdnan3 — APPROVED** [review]
LGTM

**Pull request merged**